pip/vcs/git.py

```diff
@@ -240,4 +272,19 @@ def update_submodules(self, location):
             cwd=location,
         )
 
+    @classmethod
+    def controls_location(cls, location):
+        if super(Git, cls).controls_location(location):
+            return True
+        try:
+            r = Git().run_command(['rev-parse'],
+                                  cwd=location,
+                                  show_stdout=False,
+                                  raise_on_returncode=False,
+                                  warn_on_returncode=False)
+            return not r
+        except BadCommand:
```

Comment:
Ok, I see. So this?
`logger.debug("could not determine if %s is under git control because git is not available", location)`